nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/AbstractPravegaProcessor.java

```diff
@@ -136,18 +154,36 @@ public ValidationResult validate(String subject, String input, ValidationContext
         descriptors.add(PROP_SCALE_TARGET_RATE);
         descriptors.add(PROP_SCALE_FACTOR);
         descriptors.add(PROP_SCALE_MIN_NUM_SEGMENTS);
+        descriptors.add(PROP_KEYCLOAK_JSON);
+        descriptors.add(PROP_CREATE_SCOPE);
         return descriptors;
     }
 
     ComponentLog logger;
 
     public ClientConfig getClientConfig(final ProcessContext context) {
         try {
+            ClientConfig clientConfig = null;
+            boolean createScope = new Boolean(context.getProperty(PROP_CREATE_SCOPE).getValue()).booleanValue();
             final URI controllerURI = new URI(context.getProperty(PROP_CONTROLLER).getValue());
-            final ClientConfig clientConfig = ClientConfig.builder()
-                    .controllerURI(controllerURI)
-                    .build();
+            final ClientConfig.ClientConfigBuilder clientBuilder = ClientConfig.builder().controllerURI(controllerURI);
+            if(createScope)
+            {
+                clientConfig = ClientConfig.builder()
+                        .controllerURI(controllerURI)
+                        .build();
+            }
+            else if(!createScope && (context.getProperty(PROP_KEYCLOAK_JSON).getValue() != null))
```

Comment:
What does `createScope` have to do with authentication? Just use `context.getProperty(PROP_KEYCLOAK_JSON).getValue() != null` to determine whether to use Keycloak. Do not use the `createScope` flag here.

Reply:
removed the createscope check